lib/dao.js

```diff
@@ -1477,25 +1496,34 @@ DataAccessObject.update =
 DataAccessObject.updateAll = function (where, data, options, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
 
-  assert(arguments.length >= 2, 'At least two arguments are required');
-  if (options === undefined && cb === undefined) {
+  assert(arguments.length >= 1, 'At least one argument is required');
+
+  if (data === undefined && options === undefined && cb === undefined && arguments.length === 1) {
```

Comment:
contrived case:

var where, data;
where = {customerType: 'Educational'};
// data = {discount: educationalDiscount};

// .... other stuff

Customer.updateAll(where, data)
.then(...)

The data assignment was accidentally left commented out by a developer
while they experimented with other values or something. When it is run,
instead of throwing an error or doing nothing at all, it treats where as
data and overwrites the entire dataset.

I admit, it's not Promise-related, and you can't protect every developer
from every stupid thing they do, but it's not immediately obvious what's
happened because 2 values were passed to the method...


On Thu, Feb 19, 2015 at 10:03 AM, Miroslav Bajtoš <notifications@github.com>
wrote:

> In lib/dao.js
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451#discussion_r25004522>
> :
>
> > @@ -1477,25 +1496,34 @@ DataAccessObject.update =
> >  DataAccessObject.updateAll = function (where, data, options, cb) {
> >    if (stillConnecting(this.getDataSource(), this, arguments)) return;
> >
> > -  assert(arguments.length >= 2, 'At least two arguments are required');
> > -  if (options === undefined && cb === undefined) {
> > +  assert(arguments.length >= 1, 'At least one argument is required');
> > +
> > +  if (data === undefined && options === undefined && cb === undefined && arguments.length === 1) {
>
> Can we drop arguments.length === 1? Why is that check necessary?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback-datasource-juggler/pull/451/files#r25004522>
> .
>